### ISSUE TITLE
${user.name} support

### ISSUE THREAD

**Issue by andreafalzetti** [guillaumevincent/es6-template-render#4]
Hi, thanks for this useful package.

I'm trying to handle multiple objects for example:

``` javascript
let scope = {};

scope.user = {
   name: "Andrea"
}

scope.data = {
  x: 1,
  y: 2
}
```

And I would like to:

``` javascript
const render = require("es6-template-render");
const message = "Hello ${user.name}"; // this variable comes from a yaml file
const renderedMessage = render(message, scope);
```

But apparently doesn't work as I think the package doesn't support object but only string. Am I correct? Are you considering addding the feature?

**Comment by andreafalzetti** [comment]
I looked at your code as I need a quick solution for this, I have temporary changed it to:

**index.js:**

``` js
module.exports = function (string, context) {
  var key, find, re;

  for (key in context) {
    find = '\\$\\{\\s*' + key + '\\s*\\}';
    re = new RegExp(find, 'g');
    string = string.replace(re, context[key]);

    for (subkey in context[key]) {
      find = '\\$\\{\\s*' + key + "." + subkey + '\\s*\\}';
      re = new RegExp(find, 'g');
      string = string.replace(re, context[key][subkey]);
    }
  }

  return string;
};
```

It would be nice if we could make it recursive so it prints also `${user.address.line_1}` - Could you please tell me if you are interested in doing this or shall I just fork your project? 

Thank you!

**Comment by guillaumevincent** [comment]
@andreafalzetti the solution should use a recursive function indeed. 
Feel free to pull request an implementation if you want.

Thanks

**Comment by andreafalzetti** [comment]
Hi @guillaumevincent - I was a bit short of time but I ended up doing it properly (using recursion). I also wrote a few tests, it looks working great! I hope you don't mind, but I changed the `var` to `let` as suggest with ES6.

**Comment by guillaumevincent** [comment]
tests failed
`let` doesn't exist in node 4 and node 5

@gabrielperales make es6-template-render ES5 compatible : https://github.com/guillaumevincent/es6-template-render/pull/1

**Comment by andreafalzetti** [comment]
oops! I have changed it and submitted a new pull request.

**Comment by guillaumevincent** [comment]
thank you for your contribution, fixed in 1.3.0
